### PR TITLE
Simplified Qubole Quickstart 

### PR DESCRIPTION
**Pull request by cpbhagtani**
we have simplified the process for the end user, by reducing the number of steps and inputs they needed to enter.

### CODE REVIEW

File: templates/qubole-data-lake-master.template [hunk 1]
```diff
@@ -275,113 +134,10 @@
                 "yes",
                 "no"
             ],
-            "Default": "no",
+            "Default": "yes",
```
Comment (sshvans) [inline templates/qubole-data-lake-master.template]: The option to deploy demo data should be "No" by default. If customer wants to see the demo, they can select 'Yes'.

File: templates/qubole.template [hunk 2]
```diff
@@ -188,81 +118,10 @@
                 "yes",
                 "no"
             ],
-            "Default": "no",
+            "Default": "yes",
             "Description": "Create EC2 instance with Qubole walk-through guide and load sample data into RDS",
             "Type": "String"
         },
-        "DatasetS3BucketName": {
```
Comment (sshvans) [inline templates/qubole.template]: Is dataset not needed anymore?

File: templates/qubole.template [hunk 3]
```diff
@@ -504,43 +260,6 @@
             },
             "Type": "AWS::CloudFormation::Stack"
         },
-        "QuboleBastionStack": {
```
Comment (sshvans) [inline templates/qubole.template]: Qubole bastion is not required any more?

File: templates/qubole-data-lake-master.template [hunk 4]
```diff
@@ -921,7 +356,7 @@
                     }
                 },
                 "TemplateURL": {
-                    "Fn::Sub": "https://${QSS3BucketName}.s3.amazonaws.com/${QSS3KeyPrefix}submodules/quickstart-aws-vpc/templates/aws-vpc.template"
+                    "Fn::Sub": "https://${QSS3BucketName}.s3.amazonaws.com/${QSS3KeyPrefix}templates/aws-vpc.template"
```
Comment (sshvans) [inline templates/qubole-data-lake-master.template]: The path to aws-vpc.template is incorrect. The VPC quickstart is a submodule so the path should include "submodules/"

File: templates/qubole-data-lake-master.template [hunk 5]
```diff
@@ -846,7 +302,7 @@
                     }
                 },
                 "TemplateURL": {
-                    "Fn::Sub": "https://${QSS3BucketName}.s3.amazonaws.com/${QSS3KeyPrefix}templates/qubole-data-lake.template"
+                    "Fn::Sub": "https://${QSS3BucketName}.s3.amazonaws.com/${QSS3KeyPrefix}templates/qubole.template"
```
Comment (sshvans) [inline templates/qubole-data-lake-master.template]: Is qubole-data-lake.template is not used anymore?

File: templates/qubole.template (hunk 2 above)
Comment (cpbhagtani) [reply templates/qubole.template]: Yes it is needed, but not asking it from the user and using the default value.

File: templates/qubole.template (hunk 3 above)
Comment (cpbhagtani) [reply templates/qubole.template]: Yes, we have changed the Quickstart architecture where we are now deploying in public subnets, so bastion host is not required

File: templates/qubole-data-lake-master.template (hunk 4 above)
Comment (cpbhagtani) [reply templates/qubole-data-lake-master.template]: I have modified aws-vpc.template as well, where we are not creating private subnets and netgateways any more. Do I have to check in this vpc template to submodules?

File: templates/qubole-data-lake-master.template (hunk 5 above)
Comment (cpbhagtani) [reply templates/qubole-data-lake-master.template]: Yes, the new Quickstart deploys Qubole stack and demo wizard only.

**Comment by cpbhagtani** [comment]
Following are answers your questions

There are multiple parameters w.r.t EMR, Hadoop, Spark, etc. removed. Is it not used anymore? Can
 you explain a bit what have been changed in this PR vs. the existing version of this Quick Start?

>>> These parameters are being used, but not asked from the users to make it easier for the users. Default values of these parameters are being used.

Seems like this PR updates the arch. as well. Is it correct? If yes, can you please send the updated architecture diagram?

>>>Yes, It updates deployment architecture, where we are not deploying everything in public subnets removing private subnets, basion, net gateway, ELB etc.

Please remove the unused templates and submodules
>>>I will take care of it.

ci config file and input parameter files haven't been updated based on the current updates.

>>> I will update that.

File: templates/qubole.template (hunk 3 above)
Comment (sshvans) [reply templates/qubole.template]: So you are deploying database in public subnet. What's the reason of not using private subnet for data tier and moving everything to public?

File: templates/qubole-data-lake-master.template (hunk 4 above)
Comment (sshvans) [reply templates/qubole-data-lake-master.template]: You can still use aws-vpc quickstart to create only public subnets by setting the CreatePrivateSubnets flag to false. However, as mentioned above we shouldn't be putting DB in public subnet.

File: templates/qubole-data-lake-master.template (hunk 5 above)
Comment (sshvans) [reply templates/qubole-data-lake-master.template]: So datalake is not created anymore? I would like to learn more about what Qubole stack does and who will be deploying this Quick Start? We might have to rename the quick start to remove data lake if we are not deploying data lake as part of this Quick Start.

**Comment by sshvans** [comment]
Regarding hardcoding the parameter values, I wouldn't suggest to do so. It makes the templates static and for production deployments the hardcoded values may not satisfy the needs of the customer. For example, customer may want to increase no. of HadoopNodes or use a different instance size.

**Comment by cpbhagtani** [comment]
Shivansh, purpose for removing these parameters was to make it very simple and less confusing for the user. I think majority of the users using this quickstart will be new to Qubole. However user can always change these values in Qubole clusters, these values are used for creating Qubole cluster which can be modified anytime by the user.

**Comment by sshvans** [comment]
Ok, sounds good.

**Comment by cpbhagtani** [comment]
@sshvans , I have incorporated some of the review comments in my recent commit. Please review.

Regarding, not creating db in private subnet, we chose it for simplifying the quickstart, we are not assigning this db any public ip or opening it for public. We are just opening mysql port for VPC CIDR. Moreover this db is just holding the test data (for demo wizard purpose) which is once copied to hive will not be used anymore. 
We can have a quick call to further discuss the purpose of this.

**Comment by sshvans** [comment]
Sure, let's discuss it over a call. Are you available now ?

**Comment by cpbhagtani** [comment]
How about a call on thursday morning PST? I will send out the invite

**Comment by sshvans** [comment]
Closing this as different PR has been created with updates.

**Pull request closed without merging**